portal-api/src/v1/validation/fields/issue-facility/requested-cover-start-date.js

```diff
@@ -1,12 +1,25 @@
-const moment = require('moment');
+const { isBefore, format, add, startOfDay, isAfter, endOfDay, isValid } = require('date-fns');
 const { orderNumber } = require('../../../../utils/error-list-order-number');
 const {
-  dateHasSomeValues,
   dateValidationText,
 } = require('../date');
-const { formattedTimestamp } = require('../../../facility-dates/timestamp');
 const CONSTANTS = require('../../../../constants');
 const coverDatesValidation = require('../../helpers/coverDatesValidation.helpers');
+const { hasValue } = require('../../../../utils/string');
+
+/**
+ * This mimics moment's formatting behaviour. Date-fns throws a RangeError
+ * when formatting an invalid date but moment returns the string 'Invalid date'
+ * @param {Date} date 
+ * @returns {string}
+ * 
+ * This function is probably not needed and could be removed with some refactoring
```

Comment:
I'm reluctant to delve into this within the ticket but it definitely should be noted & investigated at some point. There's a chance something unexpected could be happening when a valid month & invalid day is passed in.